sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala

```diff
@@ -386,6 +386,24 @@ class ExplainSuite extends ExplainSuiteHelper with DisableAdaptiveExecutionSuite
         Nil: _*)
   }
 
+  test("Catch and log errors when failing to write to external data source") {
```

Comment:
Why do you place this test to `ExplainSuite`? How does it rely to `explain`? If you cannot find better place, please, move it to `QueryCompilationErrorsSuite`.

Reply:
there's already a similar test in this suite, so it seemed appropriate

Reply:
Your test is not related to `explain` at all.

-1 for the changes.

Reply:
alright, moving to `QueryCompilationErrorsSuite`